### PR TITLE
add prefix user keys and user permissions on server, add set-permissions user command in immuadmin

### LINKED ISSUE #152: Add prefix for user keys in immusys
This is probably needed in order to future proof immusys db for storing and differentiating entries which do not represent users.

### LINKED ISSUE #232: User Permissions
Add Read Vs. Read/Write permissions

### LINKED ISSUE #233: immuadmin command to set user permission
immuadmin user set-permission <username> --permission read|readwrite

### PR DESCRIPTION
**Pull request by padurean**
closes #152 
closes #232 
closes #233

### CODE REVIEW

File: pkg/auth/permissions.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import "bytes"
+
+var Permissions = struct {
```
[the remark this answers is not in the document]
Comment (padurean) [reply pkg/auth/permissions.go]: Fixed.

**Comment by padurean** [comment]
@mmeloni I've also fixed the bug which was causing a _permission denied_ error when changing the password of a regular user.

**Comment by padurean** [comment]
@mmeloni Please re-review this, i've changed the implementation as we've discussed.

**Comment by mmeloni** [comment]
Check all new code and looks very good.
Only a minor. KeysPrefixes should be constants as in the rest of the project.

**Comment by padurean** [comment]
@mmeloni I've just fixed it in the latest commit (keys prefixes are now constants). Thanks!

**Pull request merged**